pints/_plot.py

```diff
@@ -0,0 +1,248 @@
+    for i in xrange(n_param):
+        axes[i].acorr(chain[:,i] - np.mean(chain[:,i]), label='p'+str(i+1))
+        axes[i].set_xlim(-0.5, 20.5)
+    axes[i].set_xlabel('Lag')
+    fig.text(0.04, 0.5, 'Autocorrelation', va='center', rotation='vertical')
+
+    return fig
+
+def prediction(chain, problem):
+    """
+    Predicted time series plot.
+
+    `chain`
+        MCMC routine returned chain
+    `problem`
+        pints.SingleSeriesProblem object
+
+    """
+    if not isinstance(problem, pints.SingleSeriesProblem):
+        raise TypeError('pints.SingleSeriesProblem type is expected')
+
+    n_sample, n_param = chain.shape
+    plot_n_sample = 1000 if n_sample>1000 else n_sample
+    times = problem.times()
+
+    # Evaluate the model for all inferred parameters
+    predicted_values = []
+    for params in chain[-plot_n_sample:,:]:
+        predicted_values.append(problem.evaluate(params[:-1]))
+    predicted_values = np.array(predicted_values)
+    mean_values = np.mean(predicted_values, axis=0)
+
+    # Plot prediction
+    fig = plt.figure(figsize=(7.5, 3.75))
+    plt.xlabel('Time')
+    plt.ylabel('Value')
+    plt.plot(times, predicted_values[0], color='#1f77b4', label='inferred series')
+    for v in predicted_values[1:]:
+        plt.plot(times, v, color='#1f77b4', alpha=0.05)
+    plt.plot(times, mean_values, color='black', lw=2, label='mean inferred')
+    plt.plot(times, problem.values(), 'o', color='#7f7f7f', ms=6.5, label='data points')
+
+    return fig
+
+def forceAspect(ax, aspect=1):
+    """
+    Set aspect.
+
+    `ax`
+        matplotlib axes handler
+    `aspect`
+        aspect to be set
+
+    """
+    im = ax.get_images()
+    ex = im[0].get_extent()
+    ax.set_aspect(abs((ex[1]-ex[0])/(ex[3]-ex[2]))/aspect)
+
+def hist2d(X, Y, ax):
+    """
+    2D histogram plot.
+
+    `X`
+        An array of the first variable
+    `Y`
+        An array of the second variable
+    `ax`
+        matplotlib axes handler
+
+    """
+    Xmin, Xmax = np.min(X), np.max(X)
+    Ymin, Ymax = np.min(Y), np.max(Y)
+    x2 = np.linspace(Xmin, Xmax, 25)
+    y2 = np.linspace(Ymin, Ymax, 25)
+    histplot = ax.hist2d(X, Y, bins=[x2,y2], normed=True, cmap=plt.cm.Blues)
+    forceAspect(ax)
+
+def kde2d(X, Y, ax):
+    """
+    2D kernel density estimation plot.
+
+    `X`
+        An array of the first variable
+    `Y`
+        An array of the second variable
+    `ax`
+        matplotlib axes handler
+
+    """
+    Xmin, Xmax = np.min(X), np.max(X)
+    Ymin, Ymax = np.min(Y), np.max(Y)
+    X1, Y1 = np.mgrid[Xmin:Xmax:100j, Ymin:Ymax:100j]
+    positions = np.vstack([X1.ravel(), Y1.ravel()])
+    values = np.vstack([X, Y])
+    kernel = stats.gaussian_kde(values)
+    Z = np.reshape(kernel(positions).T, X1.shape)
+    ax.imshow(np.rot90(Z), cmap=plt.cm.Blues, extent=[Xmin, Xmax, Ymin, Ymax])
+    ax.plot(X, Y, 'k.', markersize=2)
+    forceAspect(ax)
+
+def kde1d(X, ax):
```

Comment:
Can we make `ax` the first argument? And just change it to `axes`, no (non-standard) abbreviations in method and class names (unless it's a numerical method, like CMAES).

You could also consider merging all these methods into `graph(ax, x, y=None, kde=True)` that does 1d or 2d plots depending on `y is None`. Then you could make `force_aspect` a nested function of this one so it's hidden from users. Although, maybe keep 1d and 2d separate and call them `histogram` and `scatter` ?

Reply:
I always thought `ax` was a standard abbrev. of an item of `axes` and `axes` should be _iterable_!?

Reply:
Hmmmm. Is it a system of axes (plural of axis) that it represents though? Not sure I understand why `axes` would be iterable

Reply:
Yes you are right it is a set of axes, so axes. But I usually do something like
```
fig, axes = plt.subplots(3, 4)
for ax in axes:
    ...
```
In this case, `axes` is a set of "axes", but they usually (the way I use it) are iterable, and I will iterate it through to get one actual plot and usually call it `ax`. 

Anyway, could be my convention, not sure, but now I call them `axes` in the method. :-)